core/field_number.js

```diff
@@ -51,316 +45,267 @@ goog.require('goog.userAgent');
+   * Values are strings of the number or symbol will be added to the field text
+   * when the button is pressed.
+   * @type {Array.<string>}
+   * @const
+   */
+  // Calculator order
+  static NUMPAD_BUTTONS =
+      ['7', '8', '9', '4', '5', '6', '1', '2', '3', '.', '0', '-', ' '];
+
+  /**
+   * Src for the delete icon to be shown on the num-pad.
+   * @type {string}
+   * @const
+   */
+  static NUMPAD_DELETE_ICON = 'data:image/svg+xml;utf8,' +
+    '<svg ' +
+    'xmlns="http://www.w3.org/2000/svg" viewBox="0 0 40 40">' +
+    '<path d="M28.89,11.45H16.79a2.86,2.86,0,0,0-2,.84L9.09,1' +
+    '8a2.85,2.85,0,0,0,0,4l5.69,5.69a2.86,2.86,0,0,0,2,.84h12' +
+    '.1a2.86,2.86,0,0,0,2.86-2.86V14.31A2.86,2.86,0,0,0,28.89' +
+    ',11.45ZM27.15,22.73a1,1,0,0,1,0,1.41,1,1,0,0,1-.71.3,1,1' +
+    ',0,0,1-.71-0.3L23,21.41l-2.73,2.73a1,1,0,0,1-1.41,0,1,1,' +
+    '0,0,1,0-1.41L21.59,20l-2.73-2.73a1,1,0,0,1,0-1.41,1,1,0,' +
+    '0,1,1.41,0L23,18.59l2.73-2.73a1,1,0,1,1,1.42,1.41L24.42,20Z" fill="' +
+    Colours.numPadText + '"/></svg>';
+
+  configure_(config) {
+    super.configure_(config);
+    this.decimalAllowed_ = (typeof config.precision == 'undefined') ||
+        isNaN(config.precision) || (config.precision == 0) ||
+        (Math.floor(config.precision) != config.precision);
+    this.negativeAllowed_ = (typeof config.min == 'undefined') || isNaN(config.min) ||
+        config.min < 0;
+    this.exponentialAllowed_ = this.decimalAllowed_;
   }
-  if (this.exponentialAllowed_) {
-    pattern += "|[eE]";
-  }
-  return new RegExp(pattern);
-};
-
-/**
- * Set the constraints for this field.
- * @param {number=} opt_min Minimum number allowed.
- * @param {number=} opt_max Maximum number allowed.
- * @param {number=} opt_precision Step allowed between numbers
- */
-Blockly.FieldNumber.prototype.setConstraints_ = function(opt_min, opt_max,
-    opt_precision) {
-  this.decimalAllowed_ = (typeof opt_precision == 'undefined') ||
-      isNaN(opt_precision) || (opt_precision == 0) ||
-      (Math.floor(opt_precision) != opt_precision);
-  this.negativeAllowed_ = (typeof opt_min == 'undefined') || isNaN(opt_min) ||
-      opt_min < 0;
-  this.exponentialAllowed_ = this.decimalAllowed_;
-};
 
-/**
- * Show the inline free-text editor on top of the text and the num-pad if
- * appropriate.
- * @private
- */
-Blockly.FieldNumber.prototype.showEditor_ = function() {
-  Blockly.FieldNumber.activeField_ = this;
-  // Do not focus on mobile devices so we can show the num-pad
-  var showNumPad = this.useTouchInteraction_;
-  Blockly.FieldNumber.superClass_.showEditor_.call(this, false, showNumPad);
-
-  // Show a numeric keypad in the drop-down on touch
-  if (showNumPad) {
-    this.showNumPad_();
+  /**
+   * Show the inline free-text editor on top of the text and the num-pad if
+   * appropriate.
+   * @private
+   */
+  showEditor_(e, ) {
+    // Do not focus on mobile devices so we can show the num-pad
+    var showNumPad = Blockly.Touch.TOUCH_ENABLED;
```

Comment:
I don't think this actually has the same behavior (based on the bug bash we did last week). Currently in scratch the numpad only opens if you actually poke the field. E.g. if I click it with a mouse on my Chromebook: no numpad. If I poke it with my finger on my chromebook: numpad.

Reply:
Done.